lib/android/app/src/main/java/com/reactnativenavigation/parse/TopBarOptions.java

```diff
@@ -17,43 +17,57 @@ public static TopBarOptions parse(TypefaceLoader typefaceManager, JSONObject jso
 		TopBarOptions options = new TopBarOptions();
 		if (json == null) return options;
 
-		options.title = json.optString("title", NO_VALUE);
-		options.backgroundColor = json.optInt("backgroundColor", NO_COLOR_VALUE);
-		options.textColor = json.optInt("textColor", NO_COLOR_VALUE);
-		options.textFontSize = (float) json.optDouble("textFontSize", NO_FLOAT_VALUE);
-		options.textFontFamily = typefaceManager.getTypeFace(json.optString("textFontFamily", NO_VALUE));
-		options.hidden = NavigationOptions.BooleanOptions.parse(json.optString("hidden"));
-		options.animateHide = NavigationOptions.BooleanOptions.parse(json.optString("animateHide"));
+        options.title = json.optString("title", NO_VALUE);
+        options.backgroundColor = json.optInt("backgroundColor", NO_COLOR_VALUE);
+        options.textColor = json.optInt("textColor", NO_COLOR_VALUE);
+        options.textFontSize = (float) json.optDouble("textFontSize", NO_FLOAT_VALUE);
+        options.textFontFamily = typefaceManager.getTypeFace(json.optString("textFontFamily", NO_VALUE));
+        options.hidden = NavigationOptions.BooleanOptions.parse(json.optString("hidden"));
+        options.animateHide = NavigationOptions.BooleanOptions.parse(json.optString("animateHide"));
+        options.hideOnScroll = NavigationOptions.BooleanOptions.parse(json.optString("hideOnScroll"));
+        options.drawUnder = NavigationOptions.BooleanOptions.parse(json.optString("drawUnder"));
 		options.rightButtons = Button.parseJsonArray(json.optJSONArray("rightButtons"));
 		options.leftButtons = Button.parseJsonArray(json.optJSONArray("leftButtons"));
 
 		return options;
 	}
 
-	public String title = NO_VALUE;
-	@ColorInt public int backgroundColor = NO_COLOR_VALUE;
-	@ColorInt public int textColor = NO_COLOR_VALUE;
-	public float textFontSize = NO_FLOAT_VALUE;
-	@Nullable public Typeface textFontFamily;
-	public NavigationOptions.BooleanOptions hidden = NavigationOptions.BooleanOptions.False;
-	public NavigationOptions.BooleanOptions animateHide = NavigationOptions.BooleanOptions.False;
+    public String title = NO_VALUE;
+    @ColorInt
```

Comment:
Please inline field annotations `@ColorInt public int...`
The annotation mess is an Android fuck up, instead of using int it should have been `ColorInt` object which is unboxed into an int with a post processor or something.